### PR TITLE
Use contract's BEEFY commitment ID in relayer when possible

### PR DESCRIPTION
**Pull request by denalimarsh**
Refactored the relayer BEEFY store's test suite as part of this PR after finding a bug in the store's `GetItemByCompleteVerificationTxHash` method.

### CODE REVIEW

**Comment by musnit** [comment]
added some commits with minor fixes and to change the complete tx to actually use the nonce. i think this is good to merge now.

File: ethereum/contracts/LightClientBridge.sol [hunk 1]
```diff
@@ -83,7 +83,7 @@ contract LightClientBridge {
 
     uint256 public constant THRESHOLD_NUMERATOR = 2;
     uint256 public constant THRESHOLD_DENOMINATOR = 3;
-    uint256 public constant BLOCK_WAIT_PERIOD = 45;
+    uint256 public constant BLOCK_WAIT_PERIOD = 3;
```
Comment (philipstanislaus) [inline ethereum/contracts/LightClientBridge.sol:86]: Shall we create a ticket to reset this to a production valu? Or did we discuss with Web3 Foundation that this should be 3?

**Review by philipstanislaus — COMMENTED** [review]
Nice, thanks @denalimarsh 💪

File: ethereum/contracts/LightClientBridge.sol (hunk 1 above)
Comment (musnit) [reply ethereum/contracts/LightClientBridge.sol:86]: https://github.com/Snowfork/polkadot-ethereum/issues/426

**Pull request merged**